File: loop_transformations/sources/apply_syntax_equivalent.3.f90
! @@name:	apply_syntax_equivalent.3
! @@type:	F-free
! @@operation:	compile
! @@expect:	success
! @@version:	omp_6.0
subroutine apply_complexarg_equivalent1(A)
   implicit none
   double precision :: A(100,100)
   integer :: i, j

   !$omp tile sizes(4,5)                 &
   !$omp&   apply(grid: reverse,nothing) &
   !$omp&   apply(intratile: nothing,unroll)
   do i = 1, 100
      do j = 1, 100
         A(j,i) = A(j,i) + 1
      end do
   end do
end subroutine

subroutine apply_complexarg_equivalent2(A)
   implicit none
   double precision :: A(100,100)
   integer :: i, j, i1, j1

   !$omp reverse
   do i1 = 1, 100, 4          ! grid loop 1
      do j1 = 1, 100, 5       ! grid loop 2
         do i = i1, i1+3      ! tile loop 1
            !$omp unroll
            do j = j1, j1+4   ! tile loop 2
               A(j,i) = A(j,i) + 1
            end do
         end do
      end do
   end do
end subroutine

File: loop_transformations/sources/apply_syntax_equivalent.3.c
/*
* @@name:	apply_syntax_equivalent.3
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
void apply_complexarg_equivalent1(double A[100*100])
{
   #pragma omp tile sizes(4,5)     \
      apply(grid: reverse,nothing) \
      apply(intratile: nothing,unroll)
   for (int i = 0; i < 100; ++i)
      for (int j = 0; j < 100; ++j)
         A[i*100+j] += 1;
}

void apply_complexarg_equivalent2(double A[100*100])
{
   #pragma omp reverse
   for (int i1 = 0; i1 < 100; i1+=4)          // grid loop 1
      for (int j1 = 0; j1 < 100; j1+=5)       // grid loop 2
         for (int i = i1; i < i1+4; i+=1)     // tile loop 1
            #pragma omp unroll
            for (int j = j1; j < j1+5; j+=1)  // tile loop 2
               A[i*100+j] += 1;
}
